AdminClient/src/app/customers/dialog-body/dialog-body.component.ts

```diff
@@ -0,0 +1,100 @@
+import {Component, Inject, Input, OnInit} from '@angular/core';
+import {FormControl, FormGroup} from '@angular/forms';
+import {Customer} from '../../customer';
+import {CUSTOMERS} from "../../mock-customers";
+import {CustomersService} from '../../customers.service';
+import {CustomersComponent} from '../customers.component';
+
+@Component({
+  selector: 'app-dialog-body',
+  templateUrl: './dialog-body.component.html',
+  styleUrls: ['./dialog-body.component.scss']
+})
+export class DialogBodyComponent implements OnInit {
+
+  dialogForm = new FormGroup({
+    dialogControl1: new FormControl(''),
```

Comment:
Please give this some sort of meaningful naming what differences is dialogControl1 from dialogControl10? How will you know when debugging?

Reply:
It doesnt matter if its reused. It should be reused for forms that make sense to be re-used together otherwise your validation will be all messed up. For example new user and edit user can be grouped together yes; But then New user and new user role cant.
So example new user / edit user can have fields such as
```
 email: new FormControl('', [Validators.email, Validators.required]),
 password: new FormControl('', [Validators.required, Validators.minLength(7)]),
 address: new FormControl('', [Validators.required]),
 name: new FormControl('', [Validators.pattern([a-zA-Z ]*), Validators.required])
```
on the other hand say role form should look something like
```
 email: new FormControl('', [Validators.email, Validators.required]),
 role: new FormControl('', [Validators.validateRole), Validators.required)
```
As you can see the validations dont match.
Also assume you are debugging how will you know which field was assigned to controller1-12 etc on the other hand if it had a meaningfull name like email, role, password that would be easy to understand. 
**Re-using code is good and you should aim for it but only where it makes sense**

Reply:
In that case, the file and class should also be renamed from `dialog-body`.